pandas/core/arrays/categorical.py

```diff
@@ -1882,17 +1882,11 @@ def __getitem__(self, key):
         """
         Return an item.
         """
-        if isinstance(key, (int, np.integer)):
-            i = self._codes[key]
-            return self._box_func(i)
-
-        key = check_array_indexer(self, key)
-
-        result = self._codes[key]
-        if result.ndim > 1:
+        result = super().__getitem__(key)
+        if np.ndim(result) > 1:
+            result = result._ndarray
```

Comment:
shouldn't this be self._from_backing_data(result) ?

Reply:
what hits this case?

Reply:
> shouldn't this be self._from_backing_data(result) ?

Not until we support 2D categorical.

> what hits this case?

`cat[:, None]`

Reply:
oh i c we allow this for matplotlib compat IIRC